### PR TITLE
read MATLAB configuration file in binary mode to avoid UTF-8 encoding errors when using Python 3.6

### LINKED ISSUE #2329: Matlab fails to build with Python 3
Trying to build `matlab-2020b` with system version `Python 3.6` fails to configure the build.

The error is caused by [reading a non-ascii character](https://github.com/ComputeCanada/easybuild-easyconfigs/blob/computecanada-master/easybuild/easyconfigs/m/MATLAB/MATLAB-2020a-remove-idiotic-non-ascii-character.patch).

### PR DESCRIPTION
**Pull request by boegel**
fixes #2329

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _MATLAB-2019b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3101.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/496a879e6c935538d0dc19d92c01507f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _MATLAB-2019b.eb_ (partial log available at https://gist.github.com/99b0852b743701be10561210dbfb9a26)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3101.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/60fa285ce82aa5c037548d6b5ee44f6b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _MATLAB-2019b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3101.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/7131f10dc1b53caafc0238ed7de7063c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _MATLAB-2019b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3101.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/af951c06ce8c8ccd38b279aa1c302eed for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _MATLAB-2016b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3101.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/bed2795a69308ba8786300abbb419d75 for a full test report.

File: easybuild/easyblocks/m/matlab.py [hunk 1]
```diff
@@ -86,18 +85,24 @@ def configure_step(self):
...
         try:
-            shutil.copyfile(os.path.join(self.cfg['start_dir'], 'installer_input.txt'), self.configfile)
-            config = read_file(self.configfile)
+            copy_file(os.path.join(self.cfg['start_dir'], 'installer_input.txt'), self.configfile)
 
-            regdest = re.compile(r"^# destinationFolder=.*", re.M)
-            regagree = re.compile(r"^# agreeToLicense=.*", re.M)
-            regmode = re.compile(r"^# mode=.*", re.M)
-            reglicpath = re.compile(r"^# licensePath=.*", re.M)
+            # read file in binary mode to avoid UTF-8 encoding issues whe using Python 3,
```
Comment (bartoldeman) [inline easybuild/easyblocks/m/matlab.py:90]: typo: 's/whe/when/'

**Review by bartoldeman — CHANGES_REQUESTED** [review]
tiny typo

**Review by bartoldeman — APPROVED** [review]
LGTM

**Pull request merged**